src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -486,45 +486,48 @@ internal Task<List<RawFunctionMetadata>> SendFunctionMetadataRequest()
         }
 
         // parse metadata response into RawFunctionMetadata objects for WorkerFunctionMetadataProvider to further parse and validate
-        internal void ProcessFunctionMetadataResponses(FunctionMetadataResponses functionMetadataResponses)
+        internal void ProcessFunctionMetadataResponses(FunctionMetadataResponse functionMetadataResponse)
         {
             _workerChannelLogger.LogDebug("Received the worker function metadata response from worker {worker_id}", _workerId);
 
             var functions = new List<RawFunctionMetadata>();
 
-            foreach (var metadataResponse in functionMetadataResponses.FunctionLoadRequestsResults)
+            if (functionMetadataResponse.UseDefaultMetadataIndexing == false)
             {
-                var metadata = metadataResponse.Metadata;
-                if (metadata != null)
+                foreach (var metadata in functionMetadataResponse.FunctionMetadataResults)
                 {
-                    if (metadata.Status != null && metadata.Status.IsFailure(out Exception metadataRequestEx))
+                    if (metadata != null)
                     {
-                        _workerChannelLogger.LogError($"Worker failed to index function {metadataResponse.FunctionId}");
-                        _metadataRequestErrors[metadataResponse.FunctionId] = metadataRequestEx;
-                    }
-                    var functionMetadata = new FunctionMetadata()
-                    {
-                        FunctionDirectory = metadata.Directory,
-                        ScriptFile = metadata.ScriptFile,
-                        EntryPoint = metadata.EntryPoint,
-                        Name = metadata.Name
-                    };
+                        if (metadata.Status != null && metadata.Status.IsFailure(out Exception metadataRequestEx))
+                        {
+                            _workerChannelLogger.LogError($"Worker failed to index function {metadata.Name}");
+                            _metadataRequestErrors[metadata.Name] = metadataRequestEx;
+                        }
 
-                    functionMetadata.SetFunctionId(metadataResponse.FunctionId);
+                        var functionMetadata = new FunctionMetadata()
+                        {
+                            FunctionDirectory = metadata.Directory,
+                            ScriptFile = metadata.ScriptFile,
+                            EntryPoint = metadata.EntryPoint,
+                            Name = metadata.Name
+                        };
 
-                    var bindings = new List<string>();
-                    foreach (string binding in metadata.RawBindings)
-                    {
-                        bindings.Add(binding);
-                    }
+                        functionMetadata.SetFunctionId(metadata.Name);
```

Comment:
Not 100% sure on the FunctionId convention but in the previous design with `FunctionLoadRequests` both python and dotnet returned a new guid as the function id (instead of the function name).

[This](https://github.com/Azure/azure-functions-host/blob/431f1f87db87c2ce70d7009fcc99a1b0b7a6921d/src/WebJobs.Script/Extensions/FunctionMetadataExtensions.cs#L36) `GetFunctionId()` method in `FunctionMetadataExtensions` also returns a guid if the function id does not already exist.

Reply:
@satvu Thanks for mentioning this. This helped me in fixing some tests failure.

@fabiocav  As part of this [PR merge](https://github.com/Azure/azure-functions-language-worker-protobuf/pull/61/files#:~:text=repeated-,RpcFunctionMetadata,-function_metadata_results%20%3D%201%3B) in Protobuf repo, we replaced **FunctionLoadRequest** by **RpcFunctionMetadata**. Due to this, the *FunctionId* present in **FunctionLoadRequest** got removed, causing tests to fail and also the scenario @satvu is indicating. So, I am adding FunctionId in this [PR](https://github.com/Azure/azure-functions-host/pull/7980/files#:~:text=//%20unique%20function%20identifier,%7D) in host repo. If this looks good to you, I will also update the Protobuf repo.